docs/ldml/tr35-keyboards.md

```diff
@@ -873,338 +1027,261 @@ The `display` element describes how a character, that has come from a `keyMap/ma
-
-The `keymap/@modifier` often includes multiple combinations that match. It is not necessary (or prefered) to include all of these. Instead a minimal matching element should be used, such that exactly one keymap is matched.
-
-The following are examples of situations where the `@modifiers` and `@modifier` do not match, with a different keymap definition than above.
+_Attribute:_ `baseCharacter` (optional)
 
-| `keyMap/@modifiers` | `layer/@modifier`   |
-|---------------------|---------------------|
-| `shiftL`            | `shift` (ambiguous) |
-| `altR`              | `alt`               |
-| `shiftL?+shiftR`    | `shift`             |
+**Note:** At present, this is the only option settable in the `displayOptions`.
 
-And these do match:
+> Some scripts/languages may prefer a different base than U+25CC.
+> For Lao for example, `x` is often used as a base instead of `◌`.
+> Setting `baseCharacter="x"` (for example) is a _hint_ to the implementation which
+> requests U+25CC to be substituted with `x` on display.
+> As a hint, the implementation may ignore this option.
+>
+> **Note** that not all base characters will be suitable as bases for combining marks.
 
-| `keyMap/@modifiers` | `layer/@modifier` |
-|---------------------|-------------------|
-| `shiftL shiftR`     | `shift`           |
+* * *
 
-The use of `@modifier` as an identifier for a layer, is sufficient since it is always unique among the set of `layer` elements in a keyboard.
+### <a name="Element_layerMaps" href="#Element_layerMaps">Element: layerMaps</a>
 
-* * *
+This element represents a set of `layerMap` elements with a specific physical form factor, whether
+hardware or touch layout.
 
-### 5.14 <a name="Element_row" href="#Element_row">Element: row</a>
+> <small>
+>
+> Parents: [keyboard](#Element_keyboard)
+>
+> Children: [import](#Element_import), [layerMap](#Element_layerMap), [_special_](tr35.md#special)
+>
+> Occurrence: required, multiple
+>
+> </small>
 
-A `row` element describes the keys that are present in the row of a keyboard. `row` elements are ordered within a `layout` element with the top visual row being stored first.
+_Attribute:_ `form` (required)
 
-The row element introduces the `keyId` which may be an `ISOKey` or a `specialKey`. More formally:
+> `form` may be either `hardware` or `touch`.
```

Comment:
I note that our glossary says "physical keyboard" and "virtual keyboard" but here we use "hardware" and "touch". Hmmm.

Reply:
- [ ] glossary should be harmonized.

Reply:
harmonized.